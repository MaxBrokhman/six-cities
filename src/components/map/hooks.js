import {useEffect} from 'react';
import leaflet from 'leaflet';

const CITY = [52.38333, 4.9];
const icon = leaflet.icon({
  iconUrl: `img/pin.svg`,
  iconSize: [30, 30]
});
const ZOOM = 12;

let map;
const markers = leaflet.layerGroup();

export const useMap = (offers) => {
  useEffect(() => {
    map = leaflet.map(`map`, {
      center: CITY,
      zoom: ZOOM,
      zoomControl: false,
      marker: true
    });
    map.setView(CITY, ZOOM);
    leaflet
      .tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
        attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
      })
      .addTo(map);
  }, []);
  useEffect(() => {
    if (map) {
      markers.clearLayers();
      offers.forEach(({coords}) => {
        markers.addLayer(leaflet.marker(coords, {icon}));
      });
      markers.addTo(map);
    }
  }, [offers]);
};
