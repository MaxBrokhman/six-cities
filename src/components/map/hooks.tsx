import {useEffect} from 'react';
import leaflet from 'leaflet';

import {TOffer} from '../../reducer/types';

let map;
const markers = leaflet.layerGroup();

export const useMap = (offers: Array<TOffer>, activeCard: TOffer): void => {
  const cityLocation = [
    offers[0].city.location.latitude,
    offers[0].city.location.longitude,
  ];
  const cityZoom = offers[0].city.location.zoom;
  useEffect(() => {
    map = leaflet.map(`map`, {
      center: cityLocation,
      zoom: cityZoom,
      zoomControl: false,
      marker: true,
    });
    leaflet
      .tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
        attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
      })
      .addTo(map);
    return (): void => {
      map.remove();
    };
  }, []);
  useEffect(() => {
    if (map) {
      markers.clearLayers();
      map.setView(cityLocation, cityZoom);
      offers.forEach(({location, id}) => {
        const iconUrl = activeCard && id === activeCard.id
          ? `img/pin-active.svg`
          : `img/pin.svg`;
        const icon = leaflet.icon({
          iconUrl,
          iconSize: [30, 30]
        });
        markers.addLayer(leaflet.marker([location.latitude, location.longitude], {icon}));
      });
      markers.addTo(map);
    }
  }, [
    offers,
    activeCard,
  ]);
};
