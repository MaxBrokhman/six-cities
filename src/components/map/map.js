import React, {useEffect} from 'react';
import leaflet from 'leaflet';

const CITY = [52.38333, 4.9];
const icon = leaflet.icon({
  iconUrl: `img/pin.svg`,
  iconSize: [30, 30]
});
const ZOOM = 12;
// const offerCords = [52.3709553943508, 4.89309666406198];

const useMap = (offers) => {
  useEffect(() => {
    const map = leaflet.map(`map`, {
      center: CITY,
      zoom: ZOOM,
      zoomControl: false,
      marker: true
    });
    map.setView(CITY, ZOOM);
    leaflet
      .tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
        attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
      })
      .addTo(map);
    offers.forEach(({coords}) => {
      leaflet
      .marker(coords, {icon})
      .addTo(map);
    });
  }, []);
};
// eslint-disable-next-line
export const Map = ({offers}) => {
  useMap(offers);
  return (
    <section className="cities__map map" id="map"></section>
  );
};
