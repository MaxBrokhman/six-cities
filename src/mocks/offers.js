export const offers = [
  {
    src: `apartment-01.jpg`,
    price: 120,
    type: `Apartment`,
    description: `Beautiful &amp; luxurious apartment at great location`,
    coords: [52.3909553943508, 4.85309666406198]
  },
  {
    src: `room.jpg`,
    price: 80,
    type: `Private room`,
    description: `Wood and stone place`,
    coords: [52.369553943508, 4.85309666406198]
  },
  {
    src: `apartment-02.jpg`,
    price: 132,
    type: `Apartment`,
    description: `Canal View Prinsengracht`,
    coords: [52.3909553943508, 4.929309666406198]
  },
  {
    src: `apartment-03.jpg`,
    price: 180,
    type: `Apartment`,
    description: `Nice, cozy, warm big bed apartment`,
    coords: [52.3809553943508, 4.939309666406198]
  }
];
