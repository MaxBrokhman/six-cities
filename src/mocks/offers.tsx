import {TOffer} from "../reducer/types";


export const offers: Array<TOffer> = [
  {
    src: `img/apartment-01.jpg`,
    price: 120,
    type: `Apartment`,
    description: `Beautiful &amp; luxurious apartment at great location`,
    city: {
      location: {
        latitude: 52.3909553943508,
        longitude: 4.85309666406198,
        zoom: 6,
      },
      name: `amsterdam`,
    },
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 6,
    },
    rating: 5,
    // eslint-disable-next-line
    is_favorite: false,
    id: 1,
    // eslint-disable-next-line
    preview_image: ``,
    // eslint-disable-next-line
    is_premium: false,
    images: [],
    title: ``,
    bedrooms: 3,
    // eslint-disable-next-line
    max_adults: 3,
    goods: [],
    host: null,
  },
  {
    src: `img/room.jpg`,
    price: 80,
    type: `Private room`,
    description: `Wood and stone place`,
    city: {
      location: {
        latitude: 52.369553943508,
        longitude: 4.85309666406198,
        zoom: 6,
      },
      name: `berlin`,
    },
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 6,
    },
    rating: 4,
    // eslint-disable-next-line
    is_favorite: false,
    id: 2,
    // eslint-disable-next-line
    preview_image: ``,
    // eslint-disable-next-line
    is_premium: false,
    images: [],
    title: ``,
    bedrooms: 3,
    // eslint-disable-next-line
    max_adults: 3,
    goods: [],
    host: null,
  },
  {
    src: `img/apartment-02.jpg`,
    price: 132,
    type: `Apartment`,
    description: `Canal View Prinsengracht`,
    city: {
      location: {
        latitude: 52.369553943508,
        longitude: 4.85309666406198,
        zoom: 6,
      },
      name: `paris`,
    },
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 6,
    },
    rating: 5,
    // eslint-disable-next-line
    is_favorite: false,
    id: 3,
    // eslint-disable-next-line
    preview_image: ``,
    // eslint-disable-next-line
    is_premium: false,
    images: [],
    title: ``,
    bedrooms: 3,
    // eslint-disable-next-line
    max_adults: 3,
    goods: [],
    host: null,
  },
  {
    src: `img/apartment-03.jpg`,
    price: 180,
    type: `Apartment`,
    description: `Nice, cozy, warm big bed apartment`,
    location: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198,
      zoom: 6,
    },
    city: {
      location: {
        latitude: 52.369553943508,
        longitude: 4.85309666406198,
        zoom: 6,
      },
      name: `amsterdam`,
    },
    rating: 3,
    // eslint-disable-next-line
    is_favorite: false,
    id: 4,
    // eslint-disable-next-line
    preview_image: ``,
    // eslint-disable-next-line
    is_premium: false,
    images: [],
    title: ``,
    bedrooms: 3,
    // eslint-disable-next-line
    max_adults: 3,
    goods: [],
    host: null,
  }
];
