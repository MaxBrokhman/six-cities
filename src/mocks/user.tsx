import {TUser} from "../reducer/types";

export const user: TUser = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver`,
  // eslint-disable-next-line
  avatar_url: `#`,
  // eslint-disable-next-line
  is_pro: false,
};
