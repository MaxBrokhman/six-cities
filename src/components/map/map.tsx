import React from 'react';

import {useMap} from './hooks';
import {TOffer} from '../../reducer/types';

type TMapProps = {
  offers: Array<TOffer>;
  activeCard: TOffer;
  className: string;
}

export const Map = ({
  offers,
  activeCard,
  className,
}: TMapProps): JSX.Element => {
  useMap(offers, activeCard);
  return (
    <section className={`${className} map`} id="map"></section>
  );
};
