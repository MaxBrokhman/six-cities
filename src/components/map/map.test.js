import React from 'react';
import renderer from 'react-test-renderer';

import {offers} from '../../mocks/offers';
import {Map} from './map';

it(`Renders Map component correctly`, () => {
  const tree = renderer
                .create(
                    <Map offers={offers} />
                )
                .toJSON();
  expect(tree).toMatchSnapshot();
});
