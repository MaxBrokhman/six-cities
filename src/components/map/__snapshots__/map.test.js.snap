// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Map component correctly 1`] = `
<section
  className="cities__map map"
  id="map"
/>
`;
